extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeSubscriptionExtractor.java

```diff
@@ -68,4 +97,110 @@ public String getRelatedUrl() {
         }
         return subscriptionItems;
     }
+
+    public List<SubscriptionItem> fromZipInputStream(@Nonnull final InputStream contentInputStream)
+            throws ExtractionException {
+        final ZipInputStream zipInputStream = new ZipInputStream(contentInputStream);
+
+        try {
+            ZipEntry zipEntry;
+            while ((zipEntry = zipInputStream.getNextEntry()) != null) {
+                if (zipEntry.getName().toLowerCase().endsWith(".csv")) {
+                    try {
+                        final List<SubscriptionItem> csvItems = fromCsvInputStream(zipInputStream);
+
+                        // Return it only if it has items (it exits early if it's the wrong file format)
+                        // Otherwise try the next file
+                        if (csvItems.size() > 0) {
+                            return csvItems;
+                        }
+                    } catch (ExtractionException e) {
+                        // Ignore error and go to next file
+                        // (maybe log it?)
+                    }
+                }
+            }
+        } catch (IOException e) {
+            throw new InvalidSourceException("Error reading contents of zip file", e);
+        }
+
+        throw new InvalidSourceException("Unable to find a valid subscriptions.csv file (try extracting and selecting the csv file)");
+    }
+
+    public List<SubscriptionItem> fromCsvInputStream(@Nonnull final InputStream contentInputStream)
+            throws ExtractionException {
+        // Expected format of CSV file:
+        //      Channel Id,Channel Url,Channel Title
+        //      UC1JTQBa5QxZCpXrFSkMxmPw,http://www.youtube.com/channel/UC1JTQBa5QxZCpXrFSkMxmPw,Raycevick
+        //      UCFl7yKfcRcFmIUbKeCA-SJQ,http://www.youtube.com/channel/UCFl7yKfcRcFmIUbKeCA-SJQ,Joji
+        //
+        // Notes:
+        //      It's always 3 columns
+        //      The first line is always a header
+        //      Header names are different based on the locale
+        //      Fortunately the data is always the same order no matter what locale
+
+        int currentLine = 0;
+        String line = "";
+
+        try (BufferedReader br = new BufferedReader(new InputStreamReader(contentInputStream))) {
+            final List<SubscriptionItem> subscriptionItems = new ArrayList<>();
+
+            // Ignore header
+            currentLine = 1;
+            line = br.readLine();
```

Comment:
```suggestion
        try (BufferedReader br = new BufferedReader(new InputStreamReader(contentInputStream))) {
            final List<SubscriptionItem> subscriptionItems = new ArrayList<>();
            int currentLine = 1; // Ignore header
            String line = br.readLine();
```

Reply:
Locals currentLine and line are at the top of the function because the catch throws an exception with a detailed message using those locals.
Moving the locals to the try and I'll need to make the exception message more generic.

Knowing this, please let me know if you still want your suggested changed applied.

Reply:
Oh ok, I see, then leave it like this